checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -1664,35 +1712,112 @@ public int hashCode() {
   }
 
   /**
-   * A pair of a local variable along with a tree in the corresponding method that "assigns" the
-   * variable. Besides a normal assignment, the tree may be a {@link VariableTree} in the case of a
-   * formal parameter. We keep the tree for error-reporting purposes (so we can report an error per
-   * assignment to a local, pinpointing the expression whose MustCall may not be satisfied).
+   * An Obligation is a set of resource aliases whose must-call obligations can all be fulfilled by
+   * calling the required method(s) on any of the resource aliases. {@link
+   * MustCallConsistencyAnalyzer} is a dataflow analysis whose dataflow facts are Obligations.
+   */
+  private static class Obligation {
+
+    /** The set of resource aliases that can satisfy this obligation. */
+    public final Set<ResourceAlias> resourceAliases;
```

Comment:
I would prefer that `resourceAliases` be an `ImmutableSet<ResourceAlias>` to enforce that `Obligation` is deeply immutable.  If some code were to accidentally mutate a `resourceAliases` set in the future it could be really nasty to debug.

Reply:
That's a good point. I made the change because there were several places where we constructed a mutable set of aliases and then used a copy operation, which seemed wasteful. But your point about correctness is well-taken, and I'll change it back. Thanks for the feedback!